crates/ide/src/runnables.rs

```diff
@@ -96,9 +96,7 @@ impl Runnable {
 pub(crate) fn runnables(db: &RootDatabase, file_id: FileId) -> Vec<Runnable> {
     let sema = Semantics::new(db);
     let source_file = sema.parse(file_id);
-    source_file
-        .syntax()
-        .descendants()
+    sema.descendants_with_macros(source_file.syntax().clone())
```

Comment:
Rather than iterating the syntax, I think it makes sense to convert `file_id` to `hir::Module` and just look at the defs in scope?

Reply:
I thought about it, but the original implementation find all functions and module in all scopes. Of course I can find it recursively, but it will be little bit complicated. 

But I am okay for both approaches if you prefer.